src/App.tsx

```diff
@@ -1,61 +1,102 @@
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import React, { useState } from 'react';
+import { Todo } from './types/totos';
+
+function getUserById(userId: number) {
+  return usersFromServer.find(user => user.id === userId) || null;
+}
+
+export const todos = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUserById(todo.userId),
+}));
 
 export const App = () => {
+  const [title, setTitle] = useState('');
+  const [selectUser, setSelectUser] = useState(0);
+  const [selectUserError, setSelectUserError] = useState('');
+  const [titleError, setTitleError] = useState('');
+  const [allTodos, setAllTodos] = useState<Todo[]>(todos);
+
+  const reset = () => {
+    setTitle('');
+    setSelectUser(0);
```

Comment:
Here you can use a variable like this: const `DEFAULT_USER_ID = 0;` to avoid magic values.
```suggestion
  const [selectUser, setSelectUser] = useState(DEFAULT_USER_ID);
  const [selectUserError, setSelectUserError] = useState('');
  const [titleError, setTitleError] = useState('');
  const [allTodos, setAllTodos] = useState<Todo[]>(todos);

  const reset = () => {
    setTitle('');
    setSelectUser(DEFAULT_USER_ID);
```